dask/graph_manipulation.py

```diff
@@ -0,0 +1,435 @@
+"""Tools to modify already existing dask graphs. Unlike in :mod:`dask.optimization`, the
+output collections produced by this module are typically not functionally equivalent to
+their inputs.
+"""
+import uuid
+from typing import Callable, Hashable, Optional, Set, Tuple, TypeVar
+
+from .array import Array
+from .bag import Bag
+from .base import (
+    clone_key,
+    get_collection_name,
+    replace_name_in_key,
+    tokenize,
+    unpack_collections,
+)
+from .blockwise import blockwise
+from .core import flatten
+from .dataframe import DataFrame, Series
+from .delayed import Delayed, delayed
+from .highlevelgraph import BasicLayer, HighLevelGraph, Layer
+from .utils import ensure_dict
+
+__all__ = ("bind", "checkpoint", "clone", "drop")
+
+T = TypeVar("T")
+
+
+def drop(*collections) -> Delayed:
+    """Build a :doc:`delayed` which waits until all chunks of the input collection(s)
+    have been computed before returning None.
+
+    Arguments
+    ---------
+    collections
+        Zero or more Dask collections or nested data structures containing zero or more
+        collections
+
+    Returns
+    -------
+    :doc:`delayed` yielding None
+    """
+    collections, _ = unpack_collections(*collections)
+    if len(collections) != 1:
+        return delayed(chunks.drop)(*(drop(c) for c in collections))
+    (collection,) = collections
+
+    tok = tokenize(collection)
+    name = "drop-" + tok
+    map_name = "drop_map-" + tok
+
+    if _has_less_than_two_keys(collection):
+        # No need for a map step
+        layer = {name: (chunks.drop, collection.__dask_keys__())}
+        dsk = HighLevelGraph.from_collections(name, layer, dependencies=(collection,))
+    else:
+        # Two-step map->reduce to minimize RAM usage and data transfer over the network
```

Comment:
Just for my own understanding, we're doing a map-reduce instead of just a reduce here so that we transfer a bunch of `None`s to the reduce step instead of, say, a bunch of array chunks, right?

Reply:
correct

Reply:
clarified in the comment

Reply:
Awesome, good thinking here 👍